frappe/installer.py

```diff
@@ -406,3 +406,30 @@ def is_downgrade(sql_file_path, verbose=False):
 							print("Your site will be downgraded from Frappe {0} to {1}".format(current_version, backup_version))
 
 						return downgrade
+
+
+def validate_database_sql(path, _raise=True):
+	"""Check if file has contents and if DefaultValue table exists
+
+	Args:
+		path (str): Path of the decompressed SQL file
+		_raise (bool, optional): Raise exception if invalid file. Defaults to True.
+	"""
+	_raise = False
+	error_message = ""
+
+	if not os.path.getsize(path):
+		error_message = f"{path} is an empty file!"
+		_raise = True
+
+	if not _raise:
+		with open(path, "r") as f:
+			for line in f:
+				if 'tabDefaultValue' in line:
+					error_message = "Table `tabDefaultValue` not found in file."
+					_raise = True
+
```

Comment:
could this be a problem for large database files? Why not just check the size of database file?

Reply:
We check the validity of a restore by checking if the default value table exists. So it only makes sense to check if the table definition exists in the SQL file. You could have a 100MB SQL file but without passing the below check after restore, your migration will fail and the site will be in Broken state. 

https://github.com/frappe/frappe/blob/95d6ba0c07d22f9160eef8b2604449af051a494b/frappe/database/mariadb/setup_db.py#L97 

SIze of the files shouldn't be a problem. This reads one line at a time similar to the `is_downgrade` check. 

https://github.com/frappe/frappe/blob/42b16f81a9df96549b6dda0fda3d57bb586b22e9/frappe/installer.py#L390